plugins/linear-genome-view/src/LinearGenomeView/components/Highlight.tsx

```diff
@@ -0,0 +1,200 @@
+  },
+})
+
+interface ParsedLocStringA {
+  assemblyName: string
+  refName: string
+  start: number
+  end: number
+  reversed: boolean
+}
+
+const Highlight = observer(function Highlight({ model }: { model: LGV }) {
+  const { classes } = useStyles()
+  const [open, setOpen] = useState(false)
+  const anchorEl = useRef(null)
+  const color = useTheme().palette.quaternary?.main ?? 'goldenrod'
+
+  const session = getSession(model) as SessionWithWidgets
+
+  const dismissHighlight = () => {
+    model.setHighlight({} as ParsedLocString)
+  }
+
+  const menuItems = [
+    {
+      label: 'Dismiss highlight',
+      icon: CloseIcon,
+      onClick: () => dismissHighlight(),
+    },
+    {
+      label: 'Bookmark highlighted region',
+      icon: BookmarkIcon,
+      onClick: () => {
+        let bookmarkWidget = session.widgets.get('GridBookmark')
+        if (!bookmarkWidget) {
+          bookmarkWidget = session.addWidget(
+            'GridBookmarkWidget',
+            'GridBookmark',
+          )
+        }
+        // @ts-ignore
+        bookmarkWidget.addBookmark(model.highlight as Region)
+        dismissHighlight()
+      },
+    },
+  ]
+
+  function handleClose() {
+    setOpen(false)
+  }
+
+  if (!model.highlight) {
+    return
+  }
+
+  // coords
+  const mapCoords = (r: ParsedLocStringA) => {
+    const s = model.bpToPx({
+      refName: r.refName,
+      coord: r.start,
+    })
+    const e = model.bpToPx({
+      refName: r.refName,
+      coord: r.end,
+    })
+    return s && e
+      ? {
+          width: Math.max(Math.abs(e.offsetPx - s.offsetPx), 3),
+          left: Math.min(s.offsetPx, e.offsetPx) - model.offsetPx,
+        }
+      : undefined
+  }
+
+  const h = mapCoords(model.highlight as ParsedLocStringA)
+
+  return (
+    <>
+      {h ? (
+        <div
+          className={classes.highlight}
+          style={{
+            left: h.left,
+            width: h.width,
+            background: `${colord(color).alpha(0.35).toRgbString()}`,
+            borderLeft: `solid ${color}`,
+            borderRight: `solid ${color}`,
+          }}
+        >
+          <Tooltip title={'Highlighted from URL parameter'} arrow>
+            <IconButton ref={anchorEl} onClick={() => setOpen(true)}>
+              <LinkIcon
+                fontSize="small"
+                sx={{
+                  color: `${colord(color).darken(0.2).toRgbString()}`,
+                }}
+              />
+            </IconButton>
+          </Tooltip>
+          <Menu
+            anchorEl={anchorEl.current}
+            // anchorOrigin={{ vertical: 'bottom', horizontal: 'left' }}
+            onMenuItemClick={(_event, callback) => {
+              callback(session)
+              handleClose()
+            }}
+            open={open}
+            onClose={handleClose}
+            menuItems={menuItems}
+          />
+        </div>
+      ) : null}
+    </>
+  )
+})
+
+export const OverviewHighlight = observer(function OverviewHighlight({
```

Comment:
might suggest putting OverviewHighlight in its own file. it is debatable but i've just found that smaller single purpose files with default exports is a good style for the overall codebase